nomad/structs/structs_test.go

```diff
@@ -5036,6 +5037,47 @@ func TestAllocation_WaitClientStop(t *testing.T) {
 	}
 }
 
+func TestAllocation_DisconnectTimeout(t *testing.T) {
+	type testCase struct {
+		desc          string
+		maxDisconnect *time.Duration
+	}
+
+	testCases := []testCase{
+		{
+			desc:          "no max_client_disconnect",
+			maxDisconnect: nil,
+		},
```

Comment:
Because it's such a gotcha all the time for us, can we add a zero-value test too?

Reply:
Added in next push.